stack/stack.go

```diff
@@ -43,11 +50,21 @@ func ParseYAML(yamlFile string) (*Services, error) {
 		return nil, fmt.Errorf("'%s' is the only valid provider for this tool - found: %s", providerName, services.Provider.Name)
 	}
 
+	if len(filter) > 0 {
+		for k, function := range services.Functions {
+			function.Name = k
+			match, _ := regexp.MatchString(filter, function.Name)
```

Comment:
should handle the error here and exit with a message warning the user why the command has failed